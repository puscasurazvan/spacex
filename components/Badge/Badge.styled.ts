import styled from 'styled-components';

export const Badge = styled.h3(({ color }) => ({
  width: 'fit-content',
  padding: '5px',
  borderRadius: '5px',
  fontWeight: 'bold',
  textShadow: `0.3px 0.3px ${color}`,
  color: '#fff',
  backgroundColor: color,
  margin: '10px auto',
}));
