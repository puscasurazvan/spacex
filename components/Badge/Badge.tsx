import * as Styled from './Badge.styled';

type BadgeProps = {
  launchStatus: string;
};

export const Badge = ({ launchStatus }: BadgeProps) => (
  <Styled.Badge color={launchStatus === 'Success' ? 'green' : 'red'} data-testid="badge">
    {launchStatus}
  </Styled.Badge>
);
