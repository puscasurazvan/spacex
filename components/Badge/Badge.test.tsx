import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For additional DOM matchers
import { Badge } from './Badge'; // Replace './Badge' with the correct path to your Badge component

describe('Badge Component', () => {
  it('renders the badge with the correct color and text', () => {
    const launchStatus = 'Success';

    const { getByTestId, getByText } = render(<Badge launchStatus={launchStatus} />);
    const badge = getByTestId('badge');
    const textElement = getByText(launchStatus);

    expect(badge).toBeInTheDocument();
    expect(textElement).toBeInTheDocument();

    expect(badge).toMatchSnapshot();
  });

  it('renders the badge with red color for non-Success status', () => {
    const launchStatus = 'Failure';

    const { getByTestId } = render(<Badge launchStatus={launchStatus} />);
    const badge = getByTestId('badge');

    expect(badge).toBeInTheDocument();
  });
});
