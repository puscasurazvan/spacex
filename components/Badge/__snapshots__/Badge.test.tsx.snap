// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders the badge with the correct color and text 1`] = `
<h3
  class="sc-bczSft kqKmyZ"
  color="green"
  data-testid="badge"
>
  Success
</h3>
`;
